src/common.jl

```diff
@@ -209,6 +209,113 @@ function make_dc_powerflowdata(
     )
 end
 
+function make_bus_slack_participation_factors(
+    sys::System,
+    generator_slack_participation_factors::Dict{String, Float64},
+    bus_lookup::Dict{Int, Int},
+    time_steps::Int,
+    n_buses::Int,
+    ::Matrix{PSY.ACBusTypes},
+)
+    I = Int[]
+    J = Int[]
+    V = Float64[]
+
+    for (gen_name, val) in generator_slack_participation_factors
+        val == 0.0 && continue
+        gen = PSY.get_component(PSY.Generator, sys, gen_name)
+        isnothing(gen) && throw(ArgumentError("Generator $gen_name not found"))
+        PSY.get_available(gen) || continue
+        bus = PSY.get_bus(gen)
+        bus_idx = bus_lookup[PSY.get_number(bus)]
+        for time_step in 1:time_steps
+            push!(I, bus_idx)
+            push!(J, time_step)
+            push!(V, val)
```

Comment:
Does this correctly handle the case where multiple generators are attached to one bus?

Reply:
Yes, because of the way a sparse matrix is initialized - duplicates are summed. 

TODO: add comment in the code for clarity